sdk/src/main/java/com/ifountain/opsgenie/client/model/BaseRequest.java

```diff
@@ -72,6 +76,94 @@ public Map<String,String> getReqHeadersForGetOrDelete(){
         return addGenieKey();
     }
 
+    public boolean isValidDate(String dateStr) {
+        try {
+            Instant instantDate = Instant.parse(dateStr);
+        }
+        catch (DateTimeParseException e){
+            return false;
+        }
+        return true;
+    }
+
+    public void validateRotations(List<ScheduleRotation> rotations) throws OpsGenieClientValidationException {
+        if(Objects.nonNull(rotations)){
+            for(ScheduleRotation rotation : rotations){
+                validateScheduleRotation(rotation);
+            }
+        }
+    }
+
+    private void validateScheduleRotation(ScheduleRotation rotation) throws OpsGenieClientValidationException {
+        if(rotation.getRotationType() == null)
```

Comment:
Can we make this generic some how? It looks like too many If conditions. May be some other way to handle like creating a genric method and passing exception and value